cmd/vicadmin/logs.go

```diff
@@ -230,114 +230,123 @@ func findDiagnosticLogs(c *session.Session) (map[string]entryReader, error) {
 }
 
 func tarEntries(readers map[string]entryReader, out io.Writer) error {
-	defer trace.End(trace.Begin(""))
-
-	r, w := io.Pipe()
-	t := tar.NewWriter(w)
-
-	wg := new(sync.WaitGroup)
-	wg.Add(1)
-
-	// stream tar to out
-	go func() {
-		_, err := io.Copy(out, r)
-		if err != nil {
-			log.Errorf("error copying tar: %s", err)
-		}
-		wg.Done()
-	}()
-
-	for name, r := range readers {
-		log.Infof("Collecting log with reader %s(%#v)", name, r)
-
-		e, err := r.open()
-		if err != nil {
-			log.Warningf("error reading %s(%s): %s\n", name, r, err)
-			continue
-		}
-
-		header := tar.Header{
-			Name:    name,
-			Size:    e.Size(),
-			Mode:    0640,
-			ModTime: time.Now(),
-		}
-
-		err = t.WriteHeader(&header)
-		if err != nil {
-			log.Errorf("Failed to write header for %s: %s", header.Name, err)
-			continue
-		}
-
-		log.Infof("%s has size %d", header.Name, header.Size)
-
-		// be explicit about the number of bytes to copy as the log files will likely
-		// be written to during this exercise
-		_, err = io.CopyN(t, e, e.Size())
-		_ = e.Close()
-		if err != nil {
-			log.Errorf("Failed to write content for %s: %s", header.Name, err)
-			continue
-		}
-	}
-
-	_ = t.Flush()
-	_ = w.Close()
-	wg.Wait()
-	_ = r.Close()
-
-	return nil
+    defer trace.End(trace.Begin(""))
+
+    r, w := io.Pipe()
+    t := tar.NewWriter(w)
+
+    wg := new(sync.WaitGroup)
+    wg.Add(1)
+
+    // stream tar to out
+    go func() {
+        _, err := io.Copy(out, r)
+        if err != nil {
+            log.Errorf("error copying tar: %s", err)
+        }
+        wg.Done()
+    }()
+
+    for name, r := range readers {
+        log.Infof("Collecting log with reader %s(%#v)", name, r)
+
+        e, err := r.open()
+        if err != nil {
+            log.Warningf("error reading %s(%s): %s\n", name, r, err)
+            continue
+        }
+        var sz int64
+        if e != nil {
+            sz = e.Size()
+        }
+
+        header := tar.Header{
+            Name:    name,
+            Size:    sz,
+            Mode:    0640,
+            ModTime: time.Now(),
+        }
+
+        err = t.WriteHeader(&header)
+        if err != nil {
+            log.Errorf("Failed to write header for %s: %s", header.Name, err)
+            continue
+        }
+
+        log.Infof("%s has size %d", header.Name, header.Size)
+
+        // be explicit about the number of bytes to copy as the log files will likely
+        // be written to during this exercise
+        if e != nil {
+            _, err = io.CopyN(t, e, sz)
+            _ = e.Close()
+        }
+        if err != nil {
+            log.Errorf("Failed to write content for %s: %s", header.Name, err)
+            continue
+        }
+    }
+
+    _ = t.Flush()
+    _ = w.Close()
+    wg.Wait()
+    _ = r.Close()
+
+    return nil
```

Comment:
Why is the indent width changed? @sgairo can you run `gofmt` on these files please?